runtime/pkg/duckdbreplicator/conn.go

```diff
@@ -0,0 +1,127 @@
+package duckdbreplicator
+
+import (
+	"context"
+
+	"github.com/jmoiron/sqlx"
+)
+
+// Conn represents a single database connection.
+// This is useful when running a chain of queries using a single write connection.
+type Conn interface {
+	// Connx returns the underlying sqlx.Conn.
+	Connx() *sqlx.Conn
+
+	// CreateTableAsSelect creates a new table by name from the results of the given SQL query.
+	CreateTableAsSelect(ctx context.Context, name string, sql string, opts *CreateTableOptions) error
+
+	// InsertTableAsSelect inserts the results of the given SQL query into the table.
+	InsertTableAsSelect(ctx context.Context, name string, sql string, opts *InsertTableOptions) error
+
+	// DropTable removes a table from the database.
+	DropTable(ctx context.Context, name string) error
+
+	// RenameTable renames a table in the database.
+	RenameTable(ctx context.Context, oldName, newName string) error
+
+	// AddTableColumn adds a column to the table.
+	AddTableColumn(ctx context.Context, tableName, columnName, typ string) error
+
+	// AlterTableColumn alters the type of a column in the table.
+	AlterTableColumn(ctx context.Context, tableName, columnName, newType string) error
+}
+
+type conn struct {
+	*sqlx.Conn
+
+	db *db
+}
+
+var _ Conn = (*conn)(nil)
+
+func (c *conn) Connx() *sqlx.Conn {
+	return c.Conn
+}
+
+func (c *conn) CreateTableAsSelect(ctx context.Context, name, sql string, opts *CreateTableOptions) error {
+	if opts == nil {
+		opts = &CreateTableOptions{}
+	}
+	return c.db.createTableAsSelect(ctx, c.Conn, func() error { return nil }, name, sql, opts)
+}
+
+// InsertTableAsSelect inserts the results of the given SQL query into the table.
+func (c *conn) InsertTableAsSelect(ctx context.Context, name, sql string, opts *InsertTableOptions) error {
+	if opts == nil {
+		opts = &InsertTableOptions{
+			Strategy: IncrementalStrategyAppend,
+		}
+	}
+	return c.db.insertTableAsSelect(ctx, c.Conn, func() error { return nil }, name, sql, opts)
```

Comment:
2. Yeah `InsertTableAsSelect` can be done at both places. I kept it the package to have full set of CRUD APIs. I can evaluate `MutateTable` as well.

Reply:
If the primary concern is the `default` name, the mutation callback could just provide the inner alias to use for the table. For example:
```go
db.MutateTable(ctx, name, func(conn, alias) error {
  _, err := conn.ExecContext(ctx, fmt.Sprintf("INSERT INTO %s SELECT ...", alias))
  // ...
})
```

Alternatively, instead of `default` we could call the table by the same name as the database? That might also give nicer error messages.

I can be convinced otherwise, but if possible I think it would be nice to expose one generic mutation function that enables any calls mutation queries for two reasons:
1. We already have several mutations like append insert, merge insert, add column, and alter column, and will continuously add new kinds of mutations. It would be nice to implement these in the model executors instead of hard-coding it inside this library. 
2. We may need to expose the ability to write raw migrations or multi-statement models, which fits poorly with hardcoded functions here.

Reply:
After making the change I realise code is overall much cleaner with `MutateTable`. 
I also liked the idea of naming tables as the same as the database because otherwise the error messages have been somewhat cryptic. Also now as we are writing directly to the db file, error messages clearly say the name `table` instead of `table__db__`.default earlier.

Though it made renames slightly complicated because we can't simply rename directory names and also need to rename underlying table.